### ISSUE TITLE
Unexpected `MethodSignatureMismatch`

### ISSUE THREAD

**Issue by leighman** [vimeo/psalm#3641]
https://psalm.dev/r/cc676dadfa

The psalm type annotation works as expected but the php typehint does not.

**Comment by weirdan** [comment]
Both should be reported, as it's a fatal error: https://3v4l.org/ijRYn

**Comment by muglug** [comment]
No, only one is reported as a fatal error, appropriately

**Comment by leighman** [comment]
Maybe I'm just expected to understand PHP better but I think that difference is pretty unintuitive.

**Comment by weirdan** [comment]
Oops, I was under impression that the one with docblock was complained about. 

@leighman - docblocks are implicitly inherited (see https://psalm.dev/r/3b5275480c). As far as I understand, it's a compromise to support older (PHP5) codebases, where people tended to omit return types as they were not enforced by PHP. So the following code was fairly typical:
```php
// framework code:
interface I {
   /** @return int */
   public function method();
}

// application code:
class C implements I {
   public function method() { return 42; }
}
```
